complete/src/main/java/com/example/restservice/service/TokenService.java

```diff
@@ -35,10 +34,7 @@ public TokenDetailResponse getToken(String tokenId) {
             .findById(tokenId)
             .orElseThrow(SQInvalidRequestException::tokenNotFoundException);
     return new TokenDetailResponse(
-        tokenId,
-        token.getStatus(),
-        token.getQueue().getQueueName(),
-        getAheadCount(tokenId).orElseThrow(SQInternalServerException::new));
+        tokenId, token.getStatus(), token.getQueue().getQueueName(), getAheadCount(token));
```

Comment:
We're no longer throwing SQInternalException?

Reply:
Previously we used to
```
Optional<Long> getAheadCount() {
  if (user is removed) return Optional.empty();
}
```

and then we were checking at the caller end if the optional was empty, and then throwing TokenDeletedException.

now but it's more straightforward,

```
Long getAheadCount() {
  if (user is removed) throw TokenDeletedException();
```